File: src/main/java/pl/allegro/experiments/chi/chiserver/domain/experiments/ExperimentsRepository.java
package pl.allegro.experiments.chi.chiserver.domain.experiments;

import java.util.List;
import java.util.Optional;

public interface ExperimentsRepository {
    Optional<ExperimentDefinition> getExperiment(String experimentId);

    List<ExperimentDefinition> getAll();

    List<ExperimentDefinition> assignable();

    void delete(String experimentId);

    void save(ExperimentDefinition experiment);
}

File: src/main/kotlin/pl/allegro/experiments/chi/chiserver/domain/ExperimentsRepository.kt
package pl.allegro.experiments.chi.chiserver.domain

interface ExperimentsRepository {
    fun getExperiment(id: String): Experiment?

    val all: List<Experiment>

    fun refresh()

    fun reportingEnabled(experimentId: String): Boolean {
        val experiment = getExperiment(experimentId)
        return experiment != null && experiment.reportingEnabled
    }
}
